### DISCUSSION TITLE
What can you use to replace MinIO Gateway for GCP, Azure, and HDFS?

### DISCUSSION THREAD

**Discussion by fungiboletus** [minio/minio#15035] [Q&A] [answered]
Hei,

Minio has removed the Gateway feature for GCP, Azure, and HDFS, and I think it would be nice to discuss the alternatives.

These are the suggested solutions that have been discussed in the issue #14331 (now closed and locked 😕):

 1. Stay on an old Minio version.
    * This is not going to work in the long term. It will lack some new features, and the security vulnerabilities will accumulate over time.
 2. Use S3Proxy.
    * It's Java-based software, but it is probably the best solution today if you are on good terms with the JVM.
 3. Fork Minio and put back the gateway feature.
    * I don't think this approach is going to be successful.
 4. Stop using GCP, Azure, and HDFS as storage and fully embrace Minio.
    * You probably have good reasons not to be already doing that, but it's perhaps time to consider it again. Ceph is an alternative too.
 5. Use the Minio for AKS (Azure Kubernetes) or Minio for GKE (Google Kubernetes) when possible
    * It sounds like a good solution if you use AKS or GKE and have a budget for that:
      *  Azure or GCP infrastructure costs + $24/TB/month - minimum 100TB ($2400/month + infrastructure costs)

What are your suggestions?

**Comment by harshavardhana** [comment, marked as answer]
https://blog.min.io/deprecation-of-the-minio-gateway/ read the intentions behind why we removed gateway implementations.
